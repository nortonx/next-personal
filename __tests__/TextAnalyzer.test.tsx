import { render, screen, fireEvent } from '@testing-library/react'
import TextAnalyzer from '@/app/text-analyzer/page'

describe('TextAnalyzer', () => {
  it("Should render on screen", async () => {
    render(<TextAnalyzer />)
    const element = await screen.findByTestId('text-analyzer')
    expect(element).toBeInTheDocument()
    expect(element).toMatchSnapshot()
  })
})