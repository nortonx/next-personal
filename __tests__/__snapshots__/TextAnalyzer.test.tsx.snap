// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAnalyzer Should render on screen 1`] = `
<main
  data-testid="text-analyzer"
>
  <div
    class="container-fluid"
    data-testid="analyzer"
  >
    <div
      class="row row-cols-md-4 row-cols-1"
    >
      <div
        class="py-2"
        data-testid="counter"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="d-flex justify-content-center fs-2 card-title h5"
            >
              Words
            </div>
            <p
              class="d-flex justify-content-center text-secondary fs-3 card-text"
            >
              0
            </p>
          </div>
        </div>
      </div>
      <div
        class="py-2"
        data-testid="counter"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="d-flex justify-content-center fs-2 card-title h5"
            >
              Characters
            </div>
            <p
              class="d-flex justify-content-center text-secondary fs-3 card-text"
            >
              0
            </p>
          </div>
        </div>
      </div>
      <div
        class="py-2"
        data-testid="counter"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="d-flex justify-content-center fs-2 card-title h5"
            >
              Sentences
            </div>
            <p
              class="d-flex justify-content-center text-secondary fs-3 card-text"
            >
              0
            </p>
          </div>
        </div>
      </div>
      <div
        class="py-2"
        data-testid="counter"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="d-flex justify-content-center fs-2 card-title h5"
            >
              Paragraphs
            </div>
            <p
              class="d-flex justify-content-center text-secondary fs-3 card-text"
            >
              0
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <form
        class=""
      >
        <div
          class="mb-3"
        >
          <label
            class="fs-3 fw-semibold form-label"
          >
            Type in your text
          </label>
          <textarea
            class="form-control"
            data-testid="textarea"
            id="text-analyzer"
            rows="5"
          />
        </div>
      </form>
    </div>
  </div>
</main>
`;
